wallet/wallet.go

```diff
@@ -0,0 +1,522 @@
+package wallet
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"sync"
+	"time"
+
+	"go.sia.tech/core/chain"
+	"go.sia.tech/core/consensus"
+	"go.sia.tech/core/types"
+	"go.uber.org/zap"
+)
+
+const (
+	// transactionDefragThreshold is the number of utxos at which the wallet
+	// will attempt to defrag itself by including small utxos in transactions.
+	transactionDefragThreshold = 30
```

Comment:
Might be nice to have this configurable in a shared implementation. In case `renterd` and `hostd` want to use different values.